webapp/app/js/perf.js

```diff
@@ -28,3 +41,207 @@ perf.factory('getSeriesSummary', [ function() {
+
+    if (userOptions.pgo != undefined)
+      pgo = userOptions.pgo;
+
+    if (userOptions.targetSignature != undefined)
+      targetSignature = userOptions.targetSignature;
+
+    if (userOptions.summaryOnly != undefined)
+      summaryOnly = userOptions.summaryOnly;
+
+    return $http.get(signatureURL).then(function(response) {
+      var seriesList = [];
+      var platformList = [];
+      var testList = [];
+      var subtestSignatures = null;
+      var suiteName = "";
+
+      if (targetSignature != '') {
+          var summary = _.find(Object.keys(response.data), function(signature) { 
+                   return signature == targetSignature;
+                 });
+          if (summary) {
+            var seriesSummary = getSeriesSummary(targetSignature,
+                                                 response.data[targetSignature],
+                                                 optionMap);
+              subtestSignatures = seriesSummary.subtestSignatures;
+              suiteName = seriesSummary.name;
+          }
+      }
+
+      Object.keys(response.data).forEach(function(signature) {
+        var seriesSummary = getSeriesSummary(signature,
+                                             response.data[signature],
+                                             optionMap);
+
+        // Only keep summary signatures, filter in/out e10s and pgo
+        if (summaryOnly && !subtestSignatures && 
+            !seriesSummary.subtestSignatures ||
+            (e10s && !_.contains(seriesSummary.options, 'e10s')) ||
+            (!e10s && _.contains(seriesSummary.options, 'e10s')) ||
+            (pgo && !_.contains(seriesSummary.options, 'pgo')) ||
+            (!pgo && _.contains(seriesSummary.options, 'pgo'))) {
+            return;
+        } else {
+          // if we have input a summary signature, filter out the subtestSignatures
+          if (subtestSignatures && targetSignature) {
+            var subsig = _.find(subtestSignatures, function(subsig) { 
+                     return signature == subsig;
+                   });
+            if (subtestSignatures && !subsig) {
+              return;
+            }
+          }
+
+          seriesList.push(seriesSummary);
+
+          // add test/platform to lists if not yet present
+          if (!_.contains(platformList, seriesSummary.platform)) {
+            platformList.push(seriesSummary.platform);
+          }
+          if (!_.contains(testList, seriesSummary.name)) {
+            testList.push(seriesSummary.name);
+          }
+        }
+      });
+
+      // In the case where we pass in a summary and collect the pages, keep the test name
+      if (targetSignature != '') {
+        testList = [suiteName];
+      }
+
+      return {
+        seriesList: seriesList,
+        platformList: platformList,
+        testList: testList
+      };
+    });
+  };
+}]);
+
+
+perf.factory('getResultsMap', [ '$q', '$http', 'thServiceDomain', 'calculateStats',
```

Comment:
This method could probably also be included in the PhSeries object I propose above.